Antiban principle's added to object clicking

mrkaffeine92 opened the pull request:
-extensive antiban functions amended to _ClickHelper using  ASyncMouse; conditional to _ClickHelper as this is what's used for object interaction

40% of the time, a secondary mouse thread will initiate after object interaction [RSObject.Click/WalkClick] that will do one of the following:
1)A small mouse movement within 250 pixels of last click (min. 50 pixel)
2)A small pause (min. 1.6, max 4.0 sec) with 50% chance of moving mouse off OSRS client*
3)One of the following:
a.Random mouse movement somewhere inside screen bounds (Gauss vs Random)
b.Mouse movement onto an Object (Item, NPC, or Player)*
-followed by a 50% of right-click for (0.7, 3.0 sec) if spectacular point (object)
-20% chance of right-click on non-spectacular points (0.2, 1.0 sec)
[]followed-by moving the mouse outside Option box if right-click was successful


*: HoverMSTile: Flight, MouseOffClient: SRL.old

mrkaffeine92 committed:
Antiban principle's added to object clicking

-extensive antiban functions amended to _ClickHelper using  ASyncMouse; conditional to _ClickHelper as this is what's used for object interaction

40% of the time, a secondary mouse thread will initiate after object interaction [RSObject.Click/WalkClick] that will do one of the following:
1)A small mouse movement within 250 pixels of last click (min. 50 pixel)
2)A small pause (min. 1.6, max 4.0 sec) with 50% chance of moving mouse off OSRS client
3)One of the following:
a.Random mouse movement somewhere inside screen bounds (Gauss vs Random)
b.Mouse movement onto an Object (Item, NPC, or Player)
-followed by a 50% of right-click for (0.7, 3.0 sec) if spectacular point (object)
-20% chance of right-click on non-spectacular points (0.2, 1.0 sec)
[]followed-by moving the mouse outside Option box if right-click was successful

Torwent commented on osr/waspobject.simba:
```diff
@@ -833,10 +833,67 @@ begin
+  until(Tries > 10);
+end;
+
+procedure MouseOffClient(direction: Byte);//Copied from old SRL, modified
```
This is already sort of included in SRL with `procedure TAntiban.RandomMouse;`
I would prefer to call that instead even though it's not exactly the same.

mrkaffeine92 committed:
Timers, probability

Per Torwent's advice, probability was reduced to 20% chance of beginning antiban during object interaction.
-Timer's were also modified and I'll play more with them once I see them in the field.
-I kept my own antiban principles because I feel that there is too slight variation right now.
